Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs

```diff
@@ -189,49 +211,174 @@ public System.Collections.Generic.IEnumerable<string> RedoActionNames
+            canvas.Clear();
+            OnPropertyChanged(new PropertyChangedEventArgs("dataChanged"));
+        }
+        private void OpenFileMethod(object o)
+        {
+            try
+            {
+                if (dialogService.OpenFileDialog()) 
+                {
+                    currentFileName = dialogService.FilePath;
+                    fileService.Load(canvas, currentFileName);
+                    dialogService.ShowMessage("File is opened and data is loaded from it.");
+                }
+            }
+            catch (System.Exception ex)
+            {
+                dialogService.ShowMessage(ex.Message);
+            }
+        }
+        private void SaveFileMethod(object o)
+        {
+            try
+            {
+                if (currentFileName.Contains("Pentagon Editor"))  
+                {
+                    if (dialogService.SaveFileDialog())
+                    {
+                        currentFileName = dialogService.FilePath;
+                        fileService.Save(canvas, currentFileName);
+                        dialogService.ShowMessage("Data is saved to file.");
+                    }
+                }
+                else
+                {
+                    currentFileName = currentFileName.Remove(currentFileName.Length - 1, 1);
+                    fileService.Save(canvas, currentFileName);
+                    dialogService.ShowMessage("Data is saved to file.");
+                }
+            }
+            catch (System.Exception ex)
+            {
+                dialogService.ShowMessage(ex.Message);
+            }
+        }
+        private void SaveAsFileMethod(object o)
+        {
+            try
+            {
+                if (dialogService.SaveFileDialog())
+                {
+                    fileService.Save(canvas, dialogService.FilePath);
+                    dialogService.ShowMessage("Data is saved to file.");
+                }
+            }
+            catch (System.Exception ex)
+            {
+                dialogService.ShowMessage(ex.Message);
+            }
+        }
+        private void ExitMethod(object o)
+        {
+            if (SavingBeforeClosingWindow() != System.Windows.MessageBoxResult.Cancel) 
+            {
+                ((System.Windows.Window)o).Close();
+            }
+        }
+        private System.Windows.MessageBoxResult SavingBeforeClosingWindow()
+        {
+            if (currentFileName.Contains("*")) 
+            {
+                System.Windows.MessageBoxResult boxResult = System.Windows.MessageBox.Show("Save", "Saving", System.Windows.MessageBoxButton.YesNoCancel);
+                if (boxResult == System.Windows.MessageBoxResult.Yes) 
+                {
+                    Save();
+                }
+                return boxResult;
+            }
+            return System.Windows.MessageBoxResult.None;
+        }
+        private void Save()
+        {

```

Comment:
nice try, but it should be implemented through command, its too complex, lets get rid of it

Reply:
ok